services/collectd/service.go

```diff
@@ -211,24 +211,34 @@ func (s *Service) Open() error {
 
 // Close stops the service.
 func (s *Service) Close() error {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	if wait := func() bool {
```

Comment:
Just a thought: this anonymous function only returns `false` in one place, and with that `false` value we then return from `Close()`.  Seems like it would be fewer lines of code and the block one would be indented less if we just add the appropriate `s.mu.Unlock()` calls before the early return, and remove the anonymous function.

Reply:
The reason why I chose to do it this way was to take advantage of defer. I though there was enough going on there and enough function calls that protecting against a panic was a decent idea. Do you think I should get rid of the defer though?

Reply:
I don't mind to be honest. The way it is now involves four extra lines, while adding in two `s.mu.Unlock()` calls is only two lines. After I made that comment I noticed other cases of the same pattern where it made a lot of sense (because there were errors involved) so I think either approach would be reasonable. 👍